node/core/approval-voting/src/old_tests.rs

```diff
@@ -0,0 +1,1964 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use super::*;
+use super::approval_db::v1::Config;
+use super::backend::{Backend, BackendWriteOp};
+use polkadot_primitives::v1::{CandidateDescriptor, CoreIndex, GroupIndex, ValidatorSignature};
+use polkadot_node_primitives::approval::{
+	AssignmentCert, AssignmentCertKind, VRFOutput, VRFProof,
+	RELAY_VRF_MODULO_CONTEXT, DelayTranche,
+};
+use polkadot_node_subsystem_test_helpers::make_subsystem_context;
+use polkadot_node_subsystem::messages::AllMessages;
+use sp_core::testing::TaskExecutor;
+
+use parking_lot::Mutex;
+use bitvec::order::Lsb0 as BitOrderLsb0;
+use std::pin::Pin;
+use std::sync::Arc;
+use sp_keyring::sr25519::Keyring as Sr25519Keyring;
+use assert_matches::assert_matches;
+
+const SLOT_DURATION_MILLIS: u64 = 5000;
+
+const DATA_COL: u32 = 0;
+const NUM_COLUMNS: u32 = 1;
+
+const TEST_CONFIG: Config = Config {
+	col_data: DATA_COL,
+};
+
+fn make_db() -> DbBackend {
+	let db_writer: Arc<dyn KeyValueDB> = Arc::new(kvdb_memorydb::create(NUM_COLUMNS));
+	DbBackend::new(db_writer.clone(), TEST_CONFIG)
+}
+
+fn overlay_txn<T, F>(db: &mut T, mut f: F)
```

Comment:
`f` -> `apply_txn`?

Reply:
yessir, this is from the previous PR. Should be more descriptive... great point.